src/test/java/com/puppycrawl/tools/checkstyle/AbstractModuleTestSupport.java

```diff
@@ -372,6 +378,34 @@ private static void verifyConfig(Configuration testConfig, Configuration parsedC
         }
     }
 
+    /**
+     * Performs verification of violation lines.
+     *
+     * @param config parsed config.
+     * @param file file path.
+     * @param inputConfiguration InputConfiguration object.
+     * @throws Exception if exception occurs during verification process.
+     */
+    private void verifyViolations(Configuration config,
```

Comment:
this method does 3 things
- runs checker
- modifies list
- asserts

what if we create separate private method `getActualViolationsForFile` that will return list of violations or empty list if there are no violations for given file? 

then our `verifyViolations ` will call this newly created method, transform data and assert

Reply:
I can split. However, the thought I had in mind was to somehow combine the verify methods to run the checker only once. Right now we are running it twice, once for hardcoded violations in verify method, other for parsed violations in verifyViolations method.
I think it's okay for now to keep it this way since at the end it anyway has to do it once when we migrate to new verify method. What do you think?

Reply:
> I think it's okay for now to keep it this way since at the end it anyway has to do it once when we migrate to new verify method. What do you think?

I agree, as romani said, we should not care about performance now, it is ok run verification twice

my point was lets decompose this method into several little pieces, to make code easier to read

Reply:
done